src/main/java/core/basesyntax/suplier/FigureSupplier.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.suplier;
+
+import core.basesyntax.Figures;
+import core.basesyntax.figure.AbstractFigure;
+import core.basesyntax.figure.Circle;
+import core.basesyntax.figure.IsoscelesTrapezoid;
+import core.basesyntax.figure.Rectangle;
+import core.basesyntax.figure.RightTriangle;
+import core.basesyntax.figure.Square;
+
+public class FigureSupplier extends Supplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public AbstractFigure getRandomFigure() {
+        AbstractFigure figure = null;
+        switch (Figures.values()[Supplier.RANDOM.nextInt(Figures.values().length)]) {
+            case CIRCLE:
```

Comment:
could have been just numbers instead of enums, since it is the only place where you are using that enum, but you solution is fine too

Reply:
I made changes.